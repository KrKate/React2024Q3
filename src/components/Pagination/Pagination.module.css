button {
    padding: 3px;
    font-size: 16px;
} 

.active {
    border-color: blue;
}

.pagination {
    margin: 1rem;
    display: flex;
    gap: 0.4rem;
    justify-content: center;
    flex-wrap: wrap;
}