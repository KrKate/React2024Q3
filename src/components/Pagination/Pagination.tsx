'use client';

import Link from 'next/link';
import styles from './Pagination.module.css';
import usePaginationClient from '../../hooks/usePaginationClient';

function Pagination() {
  const { pages, currentPage, handlePageChange } = usePaginationClient();

  return (
    <div className={styles.pagination}>
      {pages.map((page) => (
        <Link
          href={`/?page=${page}`}
          key={page}
          onClick={() => handlePageChange(page)}
        >
          <button
            type="button"
            className={`${styles.paginationButton} ${currentPage === page ? styles.active : ''}`}
          >
            {page}
          </button>
        </Link>
      ))}
    </div>
  );
}

export default Pagination;
