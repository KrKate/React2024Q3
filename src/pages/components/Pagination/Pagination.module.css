.paginationButton {
    padding: 5px;
    font-size: 20px;
} 

.active {
    border-color: blue;
}

.pagination {
    margin: 1rem;
    display: flex;
    gap: 0.4rem;
    justify-content: center;
    flex-wrap: wrap;
}