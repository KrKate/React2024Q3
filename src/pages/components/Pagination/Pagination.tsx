import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import styles from './Pagination.module.css';
import { AppRootState } from '../../../redux/reducers';
import {
  setCurrentPage,
  setTotalPages,
} from '../../../redux/store/paginationSlice';

function Pagination() {
  const allProducts = useSelector(
    (state: AppRootState) => state.homePage.total
  );
  const limit = useSelector((state: AppRootState) => state.homePage.limit);
  const totalPages = useSelector(
    (state: AppRootState) => state.pagination.totalPages
  );
  const currentPage = useSelector(
    (state: AppRootState) => state.pagination.currentPage
  );
  const dispatch = useDispatch();
  // const {
  //   data: products,
  //   isLoading,
  //   isError,
  // } = useFetchPaginationQuery({ searchValue, limit, pageNumber: currentPage });

  useEffect(() => {
    dispatch(setTotalPages(Math.ceil(allProducts / limit)));
  }, [allProducts, limit, dispatch]);

  // useEffect(() => {
  //   if (products) {
  //     dispatch(setProducts(products));
  //   }
  // }, [products, dispatch]);

  const handlePageChange = (pageNumber: number) => {
    dispatch(setCurrentPage(pageNumber));
  };

  const pages = Array.from({ length: totalPages }, (_, i) => i + 1);

  return (
    <div className={styles.pagination}>
      {pages.map((page) => (
        <Link
          to={`/page=${page}`}
          key={page}
          className={currentPage === page ? styles.active : ''}
          onClick={() => handlePageChange(page)}
        >
          <button
            type="button"
            className={currentPage === page ? styles.active : ''}
          >
            {page}
          </button>
        </Link>
      ))}
    </div>
  );
}

export default Pagination;
