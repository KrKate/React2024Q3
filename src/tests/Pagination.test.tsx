import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';
import Pagination from '../components/Pagination/Pagination';

test('updates URL query parameter when page changes', () => {
  const onPageChangeMock = vi.fn();
  render(
    <MemoryRouter initialEntries={['/page=1']}>
      <Pagination
        currentPage={1}
        totalPages={5}
        onPageChange={onPageChangeMock}
      />
    </MemoryRouter>
  );
  fireEvent.click(screen.getByText('2'));
  expect(onPageChangeMock).toHaveBeenCalledWith(2);
  window.history.pushState({}, '', '?page=2');
  expect(window.location.search).toContain('page=2');
});

// describe('Pagination Component', () => {
//   it('renders correct number of pages and handles page change', () => {
//     const onPageChangeMock = vi.fn();
//     const totalPages = 5;
//     const currentPage = 3;
//     const { getByText } = render(
//       <Pagination
//         currentPage={currentPage}
//         totalPages={totalPages}
//         onPageChange={onPageChangeMock}
//       />
//     );
//     const pageButtons = Array.from({ length: totalPages }, (_, i) =>
//       getByText((i + 1).toString())
//     );

//     expect(pageButtons.length).toBe(totalPages);

//     fireEvent.click(pageButtons[1]);

//     expect(onPageChangeMock).toHaveBeenCalledWith(2);
//   });
// });
