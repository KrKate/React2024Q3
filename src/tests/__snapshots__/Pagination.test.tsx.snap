// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > check 1`] = `[Function]`;
